packages/api/models/collections.js

```diff
@@ -64,6 +66,56 @@ class Collection extends Manager {
 
     return super.create(item);
   }
+
+  /**
+   * Delete a collection
+   *
+   * @param {string} name - the collection name
+   * @param {string} version - the collection version
+   */
+  async delete(name, version) {
+    if (!(await this.exists(name, version))) throw new Error('Collection does not exist');
+
+    const associatedRuleNames = (await this.associatedRules(name, version))
+      .map((rule) => rule.name);
+
+    if (associatedRuleNames.length > 0) {
+      throw new AssociatedRulesError(
+        'Cannot delete a collection that has associated rules',
+        associatedRuleNames
+      );
+    }
+
+    await super.delete({ name, version });
+  }
+
+  /**
+   * Get any rules associated with the collection
+   *
+   * @param {string} name - collection name
+   * @param {string} version - collection version
+   * @returns {Promise<Object>}
+   */
+  async associatedRules(name, version) {
```

Comment:
Might be better named `getAssociatedRules`?

Reply:
Changed.